include/zephyr/sys/device_mmio.h

```diff
@@ -116,12 +116,12 @@ struct z_device_mmio_rom {
 
 #define Z_DEVICE_MMIO_ROM_INITIALIZER(node_id) \
 	{ \
-		.addr = DT_REG_ADDR(node_id) \
+		.addr = DT_REG_ADDR_U64(node_id) \
```

Comment:
is this always 64-bit?

Reply:
`addr` is of type `mm_reg_t` an alias for `uintptr_t`. This is not always unsigned 64-bit.

Reply:
Agree, but this shouldn't be a problem, or maybe I'm missing something?

I made this change with the following assumption:
if we assign some value to 32-bit field we expect this value to fit to 32-bits, otherwise the overflow will be detected by the compiler, and adding ULL to that integer literal will not change this behavior